modules/create_timepoint/php/endpoints/timepoint.class.inc

```diff
@@ -0,0 +1,335 @@
+                    $item = \Utility::associativeToNumericArray($item)[0];
+                    $labelOptions[$item['@']['value']] = $item['#'];
+                }
+                $visit_options[$visitLabel['@']['subprojectID']]
+                    = array_filter($labelOptions);
+            }
+        }
+        $values['visit'] = $visit_options;
+        $values['psc']   = array_intersect(
+            \Utility::getSiteList(),
+            \User::singleton()->getSiteNames()
+        );
+        if (!empty($errors)) {
+            $values['errors'] = $errors;
+        }
+        if (empty($values['psc'])) {
+            unset($values['psc']);
+        }
+        return new \LORIS\Http\Response\JsonResponse(
+            $values
+        );
+    }
+    /**
+     * Processes the values & saves to database and return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse POST request body.
+        $values = json_decode((string) $request->getBody(), true);
+        // Validate POST data.
+        $values = $this->_validate($values);
+        // User permission verification.
+        $values = $this->_verifyPermissions($request, $values);
+        if (isset($values['error'])) {
+            switch ($values['error']) {
+            case 'BadRequest':
+                return new \LORIS\Http\Response\BadRequest(
+                    'Failed to retrieve data for create timepoint.'
+                );
+            case 'Forbidden':
+                return new \LORIS\Http\Response\Forbidden(
+                    'Forbidden'
+                );
+            default:
+                break;
+            }
+        }
+        // convert site entered to a \Site object
+        $user = \User::singleton();
+        $user_list_of_sites = $user->getCenterIDs();
+        $num_sites          = count($user_list_of_sites);
+        if ($num_sites == 1) {
+            $site = \Site::singleton($user_list_of_sites[0]);
+        } else if ($num_sites > 1) {
+            $site = \Site::singleton(intval($values['psc']));
+        }
+        // List projects
+        $projects = \Utility::getProjectList();
+        $project  = null;
+        if (count($projects) == 1) {
+            //if there is only one project, autoselect first project from array of 1
+            //TODO: change this to array_key_first() when support is only PHP 7.3+
+            $project = \Project::singleton(array_pop($projects));
+        } else if (count($projects) > 1) {
+            $project_id = intval($values['psc']);
+            $project    = \Project::singleton($projects[$project_id]);
+        }
+        assert(isset($site) && $site !== null);
+        $errors   = $values['errors'];
+        $response = array();
+        if (!$errors && $site) {
```

Comment:
Can you make the check on `$site` more precise? I don't know whether a Site object in PHP has a truthy value. Also if a new object does have a True value then this part of the if statement will always evaluate to true so it's not adding anything to have it.

Reply:
@johnsaigle I'm not quite sure whats best to do here. The code is originally from the create_timepoint.class.inc _process(values) function on master. My understanding is the assert doesn't make sense here because the site object should always either be created successfully or an error would be thrown in the singleton function of the class. Should I remove the assert?

Reply:
Yes I think the assert can be removed and this line can be changed to check only for `$errors`